hazelcast/src/main/java/com/hazelcast/sql/impl/partitioner/RowPartitioner.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.partitioner;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.sql.impl.row.Row;
+
+/**
+ * Function which maps rows to partitions.
+ */
+public interface RowPartitioner extends DataSerializable {
+    /**
+     * Get partition for the row.
+     *
+     * @param row Row.
+     * @param partitionCount Number of partitions.
+     * @param serializationService Serialization service for partition calculation.
+     * @return Partition, between 0 (inclusive) and .
+     */
+    int getPartition(Row row, int partitionCount, InternalSerializationService serializationService);
```

Comment:
This method is not used in the PR because `CreateExecPlanNodeVisitor` ignores the partitioner.

Reply:
We now use SingleValuePartitioner only. This part will go away in the generalized code generation.